.github/workflows/_prs.yml

```diff
@@ -100,3 +100,41 @@ jobs:
             token: 3e6d6cb5-fcdb-41ea-b134-f6c5856363e9
           attempt_limit: 5
           attempt_delay: 10000
+
+  # you cannot use reusable workflows with an "environment" key,
+  # so we have to copy "tests-ultimate.yml" here :(
+  acceptance-tests-ultimate:
+    runs-on: ubuntu-latest
+
+    needs: [ approve ] # Require the first step to finish
+    environment:
+      name: Integrate Pull Request # Our dummy environment
+      
+    strategy:
+      max-parallel: 5
+      matrix:
+        python-version: [3.12]
+      fail-fast : false
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install test dependencies
+        run: |
+          pip install wheel
+          pip install -e .[test]
+      - name: Run acceptance tests that require Ultimate license
+        env:
+          # requested via: https://handbook.gitlab.com/handbook/marketing/developer-relations/contributor-success/community-contributors-workflows/#contributing-to-the-gitlab-enterprise-edition-ee
```

Comment:
I had done something similar when working on a gitlabform feature locally. If I'm not mistaken, this gives a temporary license that will expire soon. For CI purposes, we need something that won't expire or at least not after few weeks. This will potentially block other PRs from merging, unless we mark this as "not required". Maybe that's what we do for now? But if the license does expire, we can't confirm other changes didn't break this feature. We really need to get onto gitlab's open source program and migrate to gitlab.com.

Reply:
@amimas - agreed that would be the perfect situation, I'll address Adam's comments and also try 

Currently that is a 90/day, 100/user trial which I can repeatedly get extended -> I'll raise a ticket and talk to our Customer Support Manager, hopefully can leverage some of our weight as an Ultimate customer

Reply:
What I've done previously is use GitLab's open source programme to get Ultimate.  See https://about.gitlab.com/solutions/open-source/join/

It's a trivial process to revalidate once per year although would necessitate hosting on gitlab itself.  Could set up automated mirrors back to here though